static/js/services/lineage-model-generator.js

```diff
@@ -87,16 +87,18 @@ angular.module('inboxServices').factory('LineageModelGenerator',
       contact: function(id, options) {
         options = options || {};
         return get(id).then(function(docs) {
-          // the first row is the contact
-          var doc = docs.shift();
-           // everything else is the lineage
-          return hydrate(docs).then(function(lineage) {
-            var result = { _id: id };
+          return hydrate(docs).then(function(docs) {
```

Comment:
If i'm getting this right, this hydrating of everything including the "root" doc is new, and not tested AFAICT. (I'm relying heavily on tests for examples of what this whole file does, cos I'm not going to re-understand the code every time I have a doubt)